scrapers/youtube.js

```diff
@@ -0,0 +1,140 @@
+var request = require('sync-request');
+var cheerio = require("cheerio");
+
+const USER_AGENT = "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 " +
+	"(KHTML, like Gecko) Chrome/58.0.3029.81 Safari/537.36";
+const REQUEST_HEADER = {
+	headers: {"User-Agent": USER_AGENT}
+};
+const ENCODING = "UTF-8";
+const YOUTUBE_SEARCH_URL = "https://www.youtube.com/results?search_query=";
+const YOUTUBE_URL = "https://www.youtube.com";
+
+function getHtml(url) {
+	var result = request("GET", url, REQUEST_HEADER);
+	return result.body.toString(ENCODING);
+}
+
+function getSearchMatch(query) {
+	var url = YOUTUBE_SEARCH_URL + query.split(" ").join("+");
+	return getHtml(url);
+}
+
+function getSearchMatchVideos(html) {
+	var matchedVideoUrl = [];
+	var $ = cheerio.load(html);
+	$("a.yt-uix-tile-link").each(function(i, element){
+		matchedVideoUrl.push($(this).attr("href"));
+	});
```

Comment:
getVideoDetails uses only the first entry of this list, so  why not send the first url
something like this - ``` return $("a.yt-uix-tile-link").eq(0).attr("href"); ```